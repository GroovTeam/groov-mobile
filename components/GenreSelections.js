import React from 'react';
import { View, StyleSheet, Dimensions } from 'react-native';
import GenreSelectionButton from './GenreSelectionButton';

const window = Dimensions.get("window");
const [ windowWidth, windowHeight ] = [ window.width, window.height ];

// Styles for organizing the buttons.
const containerStyles = StyleSheet.create({
  smallPadding: {
    padding: 2,
  },
  container: {
    marginTop: 50,
    display: 'flex',
    width: windowWidth - 100,
    maxHeight: windowHeight - 500,
    justifyContent: 'flex-start',
    flexDirection: 'row',
    flexWrap: 'wrap',
    overflow: 'hidden',
  },
});

/**
 * 
 * @param {Object} data the data to render onto the buttons.
 * @param {String} color the color to render for selections.
 * @param {Callback} updateButtons update a button's status by key.
 * @returns 
 */
const GenreSelections = ({ data, color, updateButtons }) => {

  const items = []

  const toggleButton = (key) => {
    updateButtons(key);
  }

  for (const [key, value] of Object.entries(data)) {
    items.push(
    <View
      key={key}
      style={containerStyles.smallPadding}
    >
      <GenreSelectionButton
        text={key}
        color={color}
        toggleSelf={toggleButton}
        />
    </View>
    );
  }

  return (
    <View style={containerStyles.container}>
      {items}
    </View>
  );
}  

export default GenreSelections;
