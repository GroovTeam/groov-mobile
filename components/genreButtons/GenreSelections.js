import React from 'react';
import { View, StyleSheet, Dimensions } from 'react-native';
import GenreButton from './GenreButton';
import GenreSelectionButton from './GenreSelectionButton';

const window = Dimensions.get('window');
const [ windowWidth, windowHeight ] = [ window.width, window.height ];

// Styles for organizing the buttons.
const containerStyles = StyleSheet.create({
  smallPadding: {
    padding: 2,
  },
  container: {
    marginTop: 50,
    display: 'flex',
    width: windowWidth - 100,
    maxHeight: windowHeight - 500,
    justifyContent: 'flex-start',
    flexDirection: 'row',
    flexWrap: 'wrap',
    overflow: 'hidden',
  },
  staticContainer: {
    marginTop: 10,
    display: 'flex',
    width: windowWidth - 10,
    justifyContent: 'center',
    flexDirection: 'row',
    flexWrap: 'wrap',
    overflow: 'hidden',
  }
});

/**
 * 
 * @param {Object} data the data to render onto the buttons.
 * @param {String} color the color to render for selections.
 * @param {Callback} updateButtons update a button's status by key.
 * @param {Boolean} isStatic whether the buttons are a static list.
 * @returns 
 */
const GenreSelections = ({ data, color, fontSize, updateButtons, isStatic }) => {
  const items = [];

  const toggleButton = (key) => {
    updateButtons(key);
  };

  if (isStatic) {
    for (const key in data) {
      items.push(
        <View style={containerStyles.smallPadding} key={data[key]}>
          <GenreButton text={data[key]}/>
        </View>
      );
    }

    return (
      <View style={containerStyles.staticContainer}>
        {items}
      </View>
    );
  }

  for (const key of Object.keys(data)) {
    items.push(
      <View
        key={key}
        style={containerStyles.smallPadding}
      >
        <GenreSelectionButton
          text={key}
          color={color}
          fontSize={fontSize}
          toggleSelf={toggleButton}
        />
      </View>
    );
  }

  return (
    <View style={containerStyles.container}>
      {items}
    </View>
  );
};

export default GenreSelections;
