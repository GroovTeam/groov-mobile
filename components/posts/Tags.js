import React from 'react';
import { View } from 'react-native';
import GenreButton from '../genreButtons/GenreButton';

const Tags = ({ tags }) => {

  const renderableTags = [];

  tags.forEach(tag => {
    renderableTags.push(
      <View 
        style={{margin: 2}}
        key={tag}
      >
        <GenreButton
          text={tag}
          fontSize={10}
        />
      </View>
    );
  });

  return (
    <View style={{display: 'flex', flexDirection: 'row', flexWrap: 1}}>
      {renderableTags}
    </View>
  );
};

export default Tags;