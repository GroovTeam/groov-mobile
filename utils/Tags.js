export default {
  'Rap': false,
  'Opera': false,
  'Country': false,
  'Pop': false,
  'Latin': false,
  'Musical-Theatre': false,
  'Rock': false,
  'Alternative': false,
  'Hip-Hop': false,
  'K-Pop': false,
  'Samba': false,
};