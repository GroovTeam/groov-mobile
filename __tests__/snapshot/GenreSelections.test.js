import React from 'react';
import { cleanup, render } from '@testing-library/react-native';

import GenreSelections from '../../components/genreButtons/GenreSelections';
import Tags from '../../utils/Tags';

describe('<GenreSelections />', () => {
  afterEach(cleanup);

  const tags = Tags;
  const selectionColor = '#00ff0044';
  it('renders properly', () => {
    const tree = render(
      <GenreSelections
        data={tags}
        color={selectionColor}
        updateButtons={() => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});