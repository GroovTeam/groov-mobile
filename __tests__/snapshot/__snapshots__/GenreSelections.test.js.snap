// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenreSelections /> renders properly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "flex-start",
      "marginTop": 50,
      "maxHeight": 834,
      "overflow": "hidden",
      "width": 650,
    }
  }
>
  <View
    style={
      Object {
        "padding": 2,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 25,
          "borderWidth": 0.8,
          "height": "auto",
          "opacity": 1,
          "overflow": "hidden",
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(0,0,0)",
            "fontSize": 25,
            "fontWeight": "200",
            "textAlign": "center",
          }
        }
      >
        Rap
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 2,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 25,
          "borderWidth": 0.8,
          "height": "auto",
          "opacity": 1,
          "overflow": "hidden",
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(0,0,0)",
            "fontSize": 25,
            "fontWeight": "200",
            "textAlign": "center",
          }
        }
      >
        Opera
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 2,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 25,
          "borderWidth": 0.8,
          "height": "auto",
          "opacity": 1,
          "overflow": "hidden",
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(0,0,0)",
            "fontSize": 25,
            "fontWeight": "200",
            "textAlign": "center",
          }
        }
      >
        Country
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 2,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 25,
          "borderWidth": 0.8,
          "height": "auto",
          "opacity": 1,
          "overflow": "hidden",
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(0,0,0)",
            "fontSize": 25,
            "fontWeight": "200",
            "textAlign": "center",
          }
        }
      >
        Pop
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 2,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 25,
          "borderWidth": 0.8,
          "height": "auto",
          "opacity": 1,
          "overflow": "hidden",
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(0,0,0)",
            "fontSize": 25,
            "fontWeight": "200",
            "textAlign": "center",
          }
        }
      >
        Latin
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 2,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 25,
          "borderWidth": 0.8,
          "height": "auto",
          "opacity": 1,
          "overflow": "hidden",
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(0,0,0)",
            "fontSize": 25,
            "fontWeight": "200",
            "textAlign": "center",
          }
        }
      >
        Musical-Theatre
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 2,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 25,
          "borderWidth": 0.8,
          "height": "auto",
          "opacity": 1,
          "overflow": "hidden",
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(0,0,0)",
            "fontSize": 25,
            "fontWeight": "200",
            "textAlign": "center",
          }
        }
      >
        Rock
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 2,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 25,
          "borderWidth": 0.8,
          "height": "auto",
          "opacity": 1,
          "overflow": "hidden",
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(0,0,0)",
            "fontSize": 25,
            "fontWeight": "200",
            "textAlign": "center",
          }
        }
      >
        Alternative
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 2,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 25,
          "borderWidth": 0.8,
          "height": "auto",
          "opacity": 1,
          "overflow": "hidden",
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(0,0,0)",
            "fontSize": 25,
            "fontWeight": "200",
            "textAlign": "center",
          }
        }
      >
        Hip-Hop
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 2,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 25,
          "borderWidth": 0.8,
          "height": "auto",
          "opacity": 1,
          "overflow": "hidden",
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(0,0,0)",
            "fontSize": 25,
            "fontWeight": "200",
            "textAlign": "center",
          }
        }
      >
        K-Pop
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 2,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 25,
          "borderWidth": 0.8,
          "height": "auto",
          "opacity": 1,
          "overflow": "hidden",
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(0,0,0)",
            "fontSize": 25,
            "fontWeight": "200",
            "textAlign": "center",
          }
        }
      >
        Samba
      </Text>
    </View>
  </View>
</View>
`;
